jcg_testadapter_commons/src/main/scala/AdapterOptions.scala

```diff
@@ -1,10 +1,33 @@
-case class AdapterOptions(
-                           mainClass: String,
-                           classPath: Array[String],
-                           JDKPath: String,
-                           analyzeJDK: Boolean
-                         )
+class AdapterOptions(val options: Map[String, Any]) {
```

Comment:
You can make this primary constructor private, then instances can only be created via the companion object's factory methods